discovery_engine_core/ai/bert/src/pooler.rs

```diff
@@ -41,6 +44,24 @@ impl<const N: usize> From<[f32; N]> for Embedding<Ix1> {
     }
 }
 
+impl Embedding<Ix1> {
+    /// Converts from values in logical order to bytes in native byte order.
+    pub fn to_ne_bytes(&self) -> Vec<u8> {
+        self.iter().flat_map(|value| value.to_ne_bytes()).collect()
+    }
```

Comment:
If we it this way we could use a explicit byte order, like e.g. little endian which is the native endianess of probably all platforms we do support.

This would make it simpler to avoid subtle bugs if we ever add some "sync across devices" feature involving.

Reply:
i'd prefer to handle it this way, but wouldn't this break db migration? if there is a system which uses big endian as native endianess, then migrating and reading the byte blob as little endian will make a mess.

Reply:
> if there is a system which uses big endian as native 

I'm not sure if there is even a single phone currently in existence (which still is supported by us) for which this is true.

To quote [android docs](https://developer.android.com/ndk/guides/abis):

>  Android is always little-endian.

Also x86 is always little-endian. AFIK this is also true for iOs, Windows and macOs (since switching to Intel).

Arm is bi-endian but we basically compile for little endian arm (it's a bit more complex then this, but for this context we should be able to treat it as little endian).

So I'm pretty sure big endian systems do not matters for us currently.

I think likely will never matter for us, too. But if it ever does start to matter using a explicit byte order here is much less prone to bugs.

Lastly the hive->sqlite migration should go through the FFI and should use `f32` there.